service/src/main/java/bio/terra/cbas/dependencies/github/GitHubService.java

```diff
@@ -0,0 +1,61 @@
+package bio.terra.cbas.dependencies.github;
+
+import com.google.gson.Gson;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GitHubService {
+
+  private static final String BASE_URL = "https://api.github.com";
+  private final String token;
+  private final Client client;
+  private final Gson gson;
+
+  public GitHubService(String token) {
+    this.token = token;
+    this.client = ClientBuilder.newClient();
+    this.gson = new Gson();
+  }
+
+  public GetRepoResponse getRepo(String organization, String repo) {
+    WebTarget target = client.target(BASE_URL).path("/repos").path(organization).path(repo);
+    Response response =
+        target
+            .request(MediaType.APPLICATION_JSON_TYPE)
+            .header("Authorization", "token " + token)
+            .get();
+    if (response.getStatus() == 200) {
+      return gson.fromJson(response.readEntity(String.class), GetRepoResponse.class);
+    } else {
+      throw new RuntimeException("Failed to get repo: " + response.getStatusInfo());
```

Comment:
Are there different exceptions here? For example, an exception for a bad token vs. a non-existent repo.

Reply:
good call - just checked, and the API does behave differently for those examples. I'll make some changes